File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/context/XenonBeanType.java
package com.abubusoft.xenon.context;

import com.abubusoft.xenon.settings.XenonSettings;

/**
 * <p>
 * Sono i valori che possono essere trasmessi mediante reflection
 * </p>
 * 
 * @author Francesco Benincasa
 * 
 */
public enum XenonBeanType {

	/**
	 * application
	 */
	APPLICATION,

	/**
	 * 
	 */
	XENON,

	/**
	 * <p>
	 * Impostazioni caricate da file xml. il bean è di tipo
	 * {@link XenonSettings}
	 * </p>
	 * 
	 */
	XENON_SETTINGS,

	/**
	 * android context (application context)
	 */
	CONTEXT,

	/**
	 * configurazione dell'applicazione. Deve derivare da ConfigBase
	 */
	CONFIG;

	/**
	 * istanza di classe da associare
	 */
	public Object value;
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/context/XenonBeanType.kt
package com.abubusoft.xenon.context

/**
 *
 *
 * Sono i valori che possono essere trasmessi mediante reflection
 *
 *
 * @author Francesco Benincasa
 */
enum class XenonBeanType {
    /**
     * application
     */
    APPLICATION,

    /**
     *
     */
    XENON,

    /**
     *
     *
     * Impostazioni caricate da file xml. il bean è di tipo
     * [XenonSettings]
     *
     *
     */
    XENON_SETTINGS,

    /**
     * android context (application context)
     */
    CONTEXT,

    /**
     * configurazione dell'applicazione. Deve derivare da ConfigBase
     */
    CONFIG;

    /**
     * istanza di classe da associare
     */
    var value: Any? = null
}